qiskit/quantum_info/operators/symplectic/pauli_list.py

```diff
@@ -0,0 +1,1057 @@
+
+        Returns:
+            PauliList: the resulting table with the entries removed.
+
+        Raises:
+            QiskitError: if ind is out of bounds for the array size or
+                         number of qubits.
+        """
+        if isinstance(ind, int):
+            ind = [ind]
+
+        # Row deletion
+        if not qubit:
+            if max(ind) >= len(self):
+                raise QiskitError(
+                    "Indices {} are not all less than the size"
+                    " of the PauliList ({})".format(ind, len(self))
+                )
+            z = np.delete(self._z, ind, axis=0)
+            x = np.delete(self._x, ind, axis=0)
+            phase = np.delete(self._phase, ind)
+
+            return PauliList(BasePauli(z, x, phase))
+
+        # Column (qubit) deletion
+        if max(ind) >= self.num_qubits:
+            raise QiskitError(
+                "Indices {} are not all less than the number of"
+                " qubits in the PauliList ({})".format(ind, self.num_qubits)
+            )
+        z = np.delete(self._z, ind, axis=1)
+        x = np.delete(self._x, ind, axis=1)
+        # Use self.phase, not self._phase as deleting qubits can change the
+        # ZX phase convention
+        return PauliList.from_symplectic(z, x, self.phase)
+
+    def insert(self, ind, value, qubit=False):
+        """Insert Pauli's into the table.
+
+        When inserting qubits the qubit index is the same as the
+        column index of the underlying :attr:`X` and :attr:`Z` arrays.
+
+        Args:
+            ind (int): index to insert at.
+            value (PauliList): values to insert.
+            qubit (bool): if True delete qubit columns, otherwise delete
+                          Pauli rows (Default: False).
+
+        Returns:
+            PauliList: the resulting table with the entries inserted.
+
+        Raises:
+            QiskitError: if the insertion index is invalid.
+        """
+        if not isinstance(ind, int):
+            raise QiskitError("Insert index must be an integer.")
+
+        if not isinstance(value, PauliList):
+            value = PauliList(value)
+
+        # Row insertion
+        size = self._num_paulis
+        if not qubit:
+            if ind > size:
+                raise QiskitError(
+                    "Index {} is larger than the number of rows in the"
+                    " PauliList ({}).".format(ind, size)
+                )
+            base_z = np.insert(self._z, ind, value._z, axis=0)
+            base_x = np.insert(self._x, ind, value._x, axis=0)
+            base_phase = np.insert(self._phase, ind, value._phase)
+            return PauliList(BasePauli(base_z, base_x, base_phase))
+
+        # Column insertion
+        if ind > self.num_qubits:
+            raise QiskitError(
+                "Index {} is greater than number of qubits"
+                " in the PauliList ({})".format(ind, self.num_qubits)
+            )
+        # TODO: Fix phase
```

Comment:
is this TODO still needed?

Reply:
Thank you for your pointing out. I forget fixing the phase.